addons/website/static/src/builder/plugins/customize_website_plugin.js

```diff
@@ -192,9 +219,46 @@ export class CustomizeWebsitePlugin extends Plugin {
         await this.makeSCSSCusto(url, colors, nullValue);
     }, 0);
     async makeSCSSCusto(url, values, defaultValue = "null") {
-        Object.keys(values).forEach((key) => {
+        const old_values = {};
+
+        for (const key of Object.keys(values)) {
             values[key] = values[key] || defaultValue;
+            old_values[key] =
+                getComputedStyle(this.document.documentElement).getPropertyValue("--" + key) ||
+                defaultValue;
+        }
+
+        // Each first overwritten value is stored to
+        // be reverted in case of a discard.
+        // Values are grouped by scss urls to minimize
+        // the amount of backend operations necessary.
+        const original_backend_values = JSON.parse(
+            localStorage.getItem("website-customization-rollback") || "[]"
+        );
+        Object.entries(old_values).forEach((old_value) => {
+            const [key, value] = old_value;
```

Comment:
One line:
```javascript
for (const [key, value] of Object.entries(old_values)) {
```

In general, using `for (const .. of ..)` instead of `forEach` makes the code clearer IMO.